packages/teams-js/src/public/authentication.ts

```diff
@@ -296,23 +304,24 @@ export namespace authentication {
         }
         return value;
       })
-      .catch((err: Error) => {
+      .catch((err: SdkError) => {
+        const errorMessage = `Error returned, code = ${err.errorCode}, message = ${err.message}`;
         if (userRequest && userRequest.failureCallback) {
-          userRequest.failureCallback(err.message);
+          userRequest.failureCallback(errorMessage);
           return null;
         }
-        throw err;
+        throw new Error(errorMessage);
```

Comment:
Doesn't this mean we will lose the callstack associated with the original Error object?

Reply:
Oh, I see the original callstack is not actually very helpful because it is basically just from here with no extra data, carry on

Reply:
I don't think so because `SdkError` objects don't derive from `Error` and don't have a callstack property or data with them:
```
export interface SdkError {
  /**
  error code
  */
  errorCode: ErrorCode;
  /**
  Optional description for the error. This may contain useful information for web-app developers.
  This string will not be localized and is not for end-user consumption. 
  App should not depend on the string content. The exact value may change. This is only for debugging purposes.
  */
  message?: string;
}
```

I think it would be like if we caught a `string` and then threw it as a new error -- we wouldn't be losing any callstack info there because a `string` doesn't carry any such info.

Does that seem reasonable?